### PR TITLE
DOC: add required scipy version to kwarg

### LINKED ISSUE #14444: DOC/FEAT Rotation.from_rotvec documents a degrees argument which doesn't exist.
As per the title, the documentation of [`scipy.spatial.transform.Rotate.from_rotvec`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.spatial.transform.Rotation.from_rotvec.html) documents an optional kwarg `degrees : bool`. However, when I try it it doesn't work.

```python
>>> from scipy.spatial.transform import Rotation
>>> foo = Rotation.from_rotvec((0, 0, 90), degrees=True)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: from_rotvec() takes no keyword arguments
>>> foo = Rotation.from_rotvec((0, 0, 90), True)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: from_rotvec() takes exactly one argument (2 given)
```

I think the `degrees` option isn't super important since conversion between deg and rad is simple enough; however, I do think that the docs should match the code. What would be the correct fix? Updating the docs or adding the feature?

### PR DESCRIPTION
**Pull request by FirefoxMetzger**
Closes: https://github.com/scipy/scipy/issues/14444

#### What does this implement/fix?
Improves the docstring by adding the minimum scipy version required to use a new kwarg.

### CODE REVIEW

File: scipy/spatial/transform/rotation.pyx [hunk 1]
```diff
@@ -789,6 +789,8 @@ cdef class Rotation:
             A single vector or a stack of vectors, where `rot_vec[i]` gives
             the ith rotation vector.
         degrees : bool, optional
+            *New Feature:* Added in v1.7.0
+        
             If True, then the given magnitudes are assumed to be in degrees.
             Default is False.
```
Comment (tupui) [inline scipy/spatial/transform/rotation.pyx:796]:
```suggestion
        degrees : bool, optional
            *New Feature:* Added in v1.7.0
        
            If True, then the given magnitudes are assumed to be in degrees.
            Default is False.

            .. versionadded:: 1.7.0

```

**Review by tupui — CHANGES_REQUESTED** [review]
Thank you for the PR @FirefoxMetzger. Just a minor thing. This is how we add such information (special tag for Sphinx), and at the end of the description.

File: scipy/spatial/transform/rotation.pyx (hunk 1 above)
Comment (FirefoxMetzger) [reply scipy/spatial/transform/rotation.pyx:796]: Cool! I didn't know about this sphinx tag; very handy. Learn something new every day :)

File: scipy/spatial/transform/rotation.pyx [hunk 2]
```diff
@@ -788,10 +788,12 @@ cdef class Rotation:
         rotvec : array_like, shape (N, 3) or (3,)
             A single vector or a stack of vectors, where `rot_vec[i]` gives
             the ith rotation vector.
-        degrees : bool, optional
+        degrees : bool, optional
```
Comment (tupui) [inline scipy/spatial/transform/rotation.pyx:791]:
```suggestion
        degrees : bool, optional
```

**Review by tupui — APPROVED** [review]
LGTM. I will wait for the doc build and then merge. Thanks for the PR!

**Comment by tupui** [comment]
@tylerjereddy Maybe could be backported as relevant to 1.7.

**Comment by tylerjereddy** [comment]
ok, I added a backport label

**Pull request merged**